src/tools/cargotest/main.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+extern crate tempdir;
+
+use tempdir::TempDir;
+use std::env;
+use std::process::Command;
+use std::path::Path;
+
+const TEST_REPOS: &'static [(&'static str, &'static str)] = &[
+    ("https://github.com/rust-lang/cargo",
+     "ff02b156f094fb83e70acd965c83c9286411914e"),
+    ("https://github.com/retep998/winapi-rs",
```

Comment:
Winapi does have tests, but they're almost entirely "Does this function link?" and "Is this type the right size and alignment?" (Some of the latter are currently commented out due to the lack of a way to specify packing other than 1 or default).